src/number_theory/cyclotomic/primitive_roots.lean

```diff
@@ -193,6 +202,154 @@ begin
       ring_hom.eq_int_cast, int.cast_id]
 end
 
-end norm
+/-- If `is_prime_pow (n : ℕ)`, `n ≠ 2` and `irreducible (cyclotomic n K)` (in particular for
+`K = ℚ`), then the norm of `ζ - 1` is `(n : ℕ).min_fac`. -/
+lemma is_prime_pow_sub_one_norm (hn : is_prime_pow (n : ℕ)) [is_cyclotomic_extension {n} K L]
+  (hirr : irreducible (cyclotomic (n : ℕ) K)) (h : n ≠ 2) :
+  norm K (ζ - 1) = (n : ℕ).min_fac :=
+begin
+  have := (coe_lt_coe 2 _).1 (lt_of_le_of_ne (succ_le_of_lt (is_prime_pow.one_lt hn))
+    (function.injective.ne pnat.coe_injective h).symm),
+  letI hprime : fact ((n : ℕ).min_fac.prime) := ⟨min_fac_prime (is_prime_pow.ne_one hn)⟩,
+  rw [sub_one_norm_eq_eval_cyclotomic hζ this hirr],
+  nth_rewrite 0 [← is_prime_pow.min_fac_pow_factorization_eq hn],
+  obtain ⟨k, hk⟩ : ∃ k, ((n : ℕ).factorization) (n : ℕ).min_fac = k + 1 := exists_eq_succ_of_ne_zero
+      (((n : ℕ).factorization.mem_support_to_fun (n : ℕ).min_fac).1
+      $ factor_iff_mem_factorization.2 $ (mem_factors (is_prime_pow.ne_zero hn)).2 ⟨hprime.out,
+      min_fac_dvd _⟩),
+  simp [hk, sub_one_norm_eq_eval_cyclotomic hζ this hirr],
+end
+
+omit hζ
+
+/-- If `irreducible (cyclotomic (p ^ (k + 1)) K)` (in particular for `K = ℚ`) and `p` is an odd
+prime, then the norm of `ζ - 1` is `p`. -/
+lemma prime_ne_two_pow.sub_one_norm {p : ℕ+} [ne_zero ((p : ℕ) : K)] (k : ℕ)
```

Comment:
```suggestion
lemma sub_one_norm.pow_prime {p : ℕ+} [ne_zero ((p : ℕ) : K)] (k : ℕ)
```

I don't think you need to say that you don't accept 2 here, if anything in the docstring and link to the other decl